features/events/traps/src/test/java/org/opennms/netmgt/trapd/TrapdIT.java

```diff
@@ -283,6 +285,70 @@ public void testSnmpV2cTrapWithAddressFromVarbind() throws Exception {
+    SnmpObjId trapOID = SnmpObjId.get(enterpriseId, new SnmpInstId(1));
+    SnmpTrapBuilder pdu = SnmpUtils.getV2TrapBuilder();
+    pdu.addVarBind(SnmpObjId.get(".1.3.6.1.2.1.1.3.0"), SnmpUtils.getValueFactory().getTimeTicks(0));
+    pdu.addVarBind(SnmpObjId.get(".1.3.6.1.6.3.1.1.4.1.0"), SnmpUtils.getValueFactory().getObjectId(trapOID));
+    pdu.addVarBind(SnmpObjId.get(".1.3.6.1.6.3.1.1.4.3.0"), SnmpUtils.getValueFactory().getObjectId(enterpriseId));
+    // The varbind with the address
+    pdu.addVarBind(TrapUtils.SNMP_TRAP_ADDRESS_OID, SnmpUtils.getValueFactory().getIpAddress(InetAddress.getByName("10.255.1.1")));
+
+    EventBuilder defaultTrapBuilder = new EventBuilder("uei.opennms.org/default/trap", "trapd");
+    defaultTrapBuilder.setInterface(remoteAddr);
+    defaultTrapBuilder.setSnmpVersion("v2c");
+    m_mockEventIpcManager.getEventAnticipator().anticipateEvent(defaultTrapBuilder.getEvent());
+
+    EventBuilder newSuspectBuilder = new EventBuilder(EventConstants.NEW_SUSPECT_INTERFACE_EVENT_UEI, "trapd");
+    // The address in the newSuspect event should match the one specified in the varbind
+    newSuspectBuilder.setInterface(remoteAddr);
+    m_mockEventIpcManager.getEventAnticipator().anticipateEvent(newSuspectBuilder.getEvent());
+
+    pdu.send(localhost, m_trapdConfig.getSnmpTrapPort(), "public");
+
+
+    InetAddress secondaryRemoteAddr = InetAddress.getByName("10.255.1.2");
+
+    SnmpObjId secondaryEnterpriseId = SnmpObjId.get(".1.3.6.1.4.1.5813");
+    SnmpObjId secondaryTrapOID = SnmpObjId.get(secondaryEnterpriseId, new SnmpInstId(1));
+    SnmpTrapBuilder secondaryPdu = SnmpUtils.getV2TrapBuilder();
+    secondaryPdu.addVarBind(SnmpObjId.get(".1.3.6.1.2.1.1.3.0"), SnmpUtils.getValueFactory().getTimeTicks(0));
+    secondaryPdu.addVarBind(SnmpObjId.get(".1.3.6.1.6.3.1.1.4.1.0"), SnmpUtils.getValueFactory().getObjectId(secondaryTrapOID));
+    secondaryPdu.addVarBind(SnmpObjId.get(".1.3.6.1.6.3.1.1.4.3.0"), SnmpUtils.getValueFactory().getObjectId(secondaryEnterpriseId));
+// The varbind with the address
+    secondaryPdu.addVarBind(TrapUtils.SNMP_TRAP_ADDRESS_OID, SnmpUtils.getValueFactory().getIpAddress(InetAddress.getByName("10.255.1.2")));
+
+    EventBuilder secondaryTrapBuilder = new EventBuilder("uei.opennms.org/default/trap", "trapd");
+    secondaryTrapBuilder.setInterface(secondaryRemoteAddr);
+    secondaryTrapBuilder.setSnmpVersion("v2c");
+    m_mockEventIpcManager.getEventAnticipator().anticipateEvent(secondaryTrapBuilder.getEvent());
+
+    EventBuilder secondarySuspectBuilder = new EventBuilder(EventConstants.NEW_SUSPECT_INTERFACE_EVENT_UEI, "trapd");
+// The address in the secondarySuspect event should match the one specified in the varbind
+    secondarySuspectBuilder.setInterface(secondaryRemoteAddr);
+    m_mockEventIpcManager.getEventAnticipator().anticipateEvent(secondarySuspectBuilder.getEvent());
+    secondaryPdu.send(localhost, m_trapdConfig.getSnmpTrapPort(), "public");
```

Comment:
i would prefer both pdus to be sent in one go, one after another to be very precise on timing.